res/controllers/common-hid-packet-parser.js

```diff
@@ -1679,7 +1679,12 @@ class HIDController {
      */
     processIncomingPacket(packet, delta) {
 
-        HIDController.fastForIn(delta, (field_name) => {
+        // Fast loop implementation over delta object
+        const fieldKeyArr = Object.keys(delta);
+        let fieldKeyIdx = fieldKeyArr.length;
+        while (fieldKeyIdx--) {
+            const field_name = fieldKeyArr[fieldKeyIdx];
```

Comment:
> The hid-parser optimization with the FastForIn wrapper did not work in one case (this object was not set correct). I reverted this to the last version without the wrapper

I think this is related to some qt bug (I don't remember the number, nor could I find the discussion on zulip anymore). But using an function expression and calling `.bind(this)` on it worked as a workaround. Have you tried that?





```suggestion
        HIDController.fastForIn(delta, (function(field_name) {....}).bind(this));
```

Reply:
I would gladly hardware test the proposed change but I'm not sure how to properly formulate the function expression here.

Reply:
this would be the exact patch I'm proposing. Are you comfortable enough with git to know how to apply it?

```patch
diff --git a/res/controllers/common-hid-packet-parser.js b/res/controllers/common-hid-packet-parser.js
index 9d1b3cc9c8..ac9ad684ff 100644
--- a/res/controllers/common-hid-packet-parser.js
+++ b/res/controllers/common-hid-packet-parser.js
@@ -1679,11 +1679,7 @@ class HIDController {
      */
     processIncomingPacket(packet, delta) {
 
-        // Fast loop implementation over delta object
-        const fieldKeyArr = Object.keys(delta);
-        let fieldKeyIdx = fieldKeyArr.length;
-        while (fieldKeyIdx--) {
-            const field_name = fieldKeyArr[fieldKeyIdx];
+         HIDController.fastForIn(delta, (function(field_name) {
 
             // @ts-ignore ignoredControlChanges should be defined in the users mapping
             // see EKS-Otus.js for an example
@@ -1702,7 +1698,7 @@ class HIDController {
             } else {
                 console.warn(`HIDController.processIncomingPacket - Unknown field ${field.name} type ${field.type}`);
             }
-        }
+        }).bind(this));
     }
     /**
      * Get active group for this field

```

Reply:
Yeah sure, I can confirm that the following implementation works as expected with both the F1 and the X1mkII:
```js
    processIncomingPacket(packet, delta) {

        HIDController.fastForIn(delta, (function(field_name) {

            // @ts-ignore ignoredControlChanges should be defined in the users mapping
            // see EKS-Otus.js for an example
            if (this.ignoredControlChanges !== undefined &&
                    // @ts-ignore
                    this.ignoredControlChanges.indexOf(field_name) !== -1) {
                return; // continue loop - by returning to fastForIn
            }
            const field = delta[field_name];
            if (field.type === "button") {
                // Button/Boolean field
                this.processButton(field);
            } else if (field.type === "control") {
                // Numeric value field
                this.processControl(field);
            } else {
                console.warn(`HIDController.processIncomingPacket - Unknown field ${field.name} type ${field.type}`);
            }
        }).bind(this));
    }
```

Reply:
great. @JoergAtGithub can you implement that (along with a comment referencing https://bugreports.qt.io/browse/QTBUG-95677)? Then we can merge.